cmd/stolonctl/cmd/clusterdata.go

```diff
@@ -67,3 +97,68 @@ func clusterdata(cmd *cobra.Command, args []string) {
 	}
 	stdout("%s", clusterdataj)
 }
+
+func isSafeToWriteClusterData(store store.Store) error {
+	if cd, _, err := store.GetClusterData(context.TODO()); err != nil {
+		return err
+	} else if cd != nil {
+		if !writeClusterdataOpts.forceYes {
+			return fmt.Errorf("WARNING: cluster data already available use --yes to override")
+		} else {
+			stdout("WARNING: The current cluster data will be removed")
+		}
+	}
+	return nil
+}
+
+func clusterData(data []byte) (*cluster.ClusterData, error) {
+	cd := cluster.ClusterData{}
+	err := json.Unmarshal(data, &cd)
+	return &cd, err
+}
+
+func writeClusterdata(reader io.Reader, s store.Store) error {
+	data, err := ioutil.ReadAll(reader)
+	if err != nil {
+		return fmt.Errorf("error while reading data: %v", err)
+	}
+
+	cd, err := clusterData(data)
+
+	if err != nil {
+		return fmt.Errorf("invalid cluster data: %v", err)
+	}
+
+	if err = isSafeToWriteClusterData(s); err != nil {
+		return fmt.Errorf("%v", err)
```

Comment:
Curious to know why can't we just return an `err` back here ?

Reply:
@rShetty Thanks for pointing it out. It was by mistake. No particular reason.